lib/macho/macho_file.rb

```diff
@@ -337,11 +346,18 @@ def linked_dylibs
     #  file.change_install_name("/usr/lib/libWhatever.dylib", "/usr/local/lib/libWhatever2.dylib")
     # @param old_name [String] the shared library's old name
     # @param new_name [String] the shared library's new name
+    # @param options [Hash]
+    # @option options [Boolean] :strict (true) whether or not to fail loudly
+    #  with an exception if the change cannot be performed
     # @return [void]
     # @raise [MachO::DylibUnknownError] if no shared library has the old name
-    def change_install_name(old_name, new_name)
+    def change_install_name(old_name, new_name, options = {})
       old_lc = dylib_load_commands.find { |d| d.name.to_s == old_name }
-      raise DylibUnknownError.new(old_name) if old_lc.nil?
+
+      if old_lc.nil?
+        return unless options.fetch(:strict, true)
+        raise DylibUnknownError.new(old_name)
+      end
```

Comment:
This change needs to be reverted. The modification methods in `MachOFile` need to always raise an exception, independently of `options` (because `options` currently only matters if multiple Mach-O slices are involved).